Form/DataField/JsonMenuEditorFieldType.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace EMS\CoreBundle\Form\DataField;
+
+use EMS\CoreBundle\Form\Field\AnalyzerPickerType;
+use EMS\CoreBundle\Form\Field\IconPickerType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class JsonMenuEditorFieldType extends DataFieldType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getLabel()
+    {
+        return 'JSON menu editor field';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getIcon()
+    {
+        return 'fa fa-sitemap';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParent()
+    {
+        return HiddenType::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $fieldType = $options['metadata'];
+
+        /*get options for twig context*/
+        parent::buildView($view, $form, $options);
+        $view->vars ['icon'] = $options ['icon'];
```

Comment:
Why not build the $attr first. Then assign $view->vars = \array_merge($view->vars, $attr) ? This way you only have one line setting data in this public variable (easier to maintain if one day this vars property becomes private for php.

Reply:
Oops should be \array_merge($view->vars, [‘attr’ => $attr, ‘icon’ => $options[‘icon’]])

Reply:
Fixed